src/test/java/org/opensearch/knn/index/codec/KNN920Codec/KNN920CodecTests.java

```diff
@@ -5,18 +5,95 @@
 
 package org.opensearch.knn.index.codec.KNN920Codec;
 
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.FieldType;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.index.SerialMergeScheduler;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.tests.index.RandomIndexWriter;
+import org.opensearch.index.mapper.MapperService;
+import org.opensearch.knn.common.KNNConstants;
+import org.opensearch.knn.index.SpaceType;
+import org.opensearch.knn.index.VectorField;
 import org.opensearch.knn.index.codec.KNNCodecTestCase;
+import org.opensearch.knn.index.mapper.KNNVectorFieldMapper;
+import org.opensearch.knn.index.memory.NativeMemoryLoadStrategy;
+import org.opensearch.knn.index.util.KNNEngine;
+import org.opensearch.watcher.ResourceWatcherService;
 
 import java.io.IOException;
+import java.util.Optional;
 import java.util.concurrent.ExecutionException;
 
+import static org.mockito.Mockito.mock;
+import static org.opensearch.knn.index.codec.KNNCodecFactory.CodecDelegateFactory.createKNN92DefaultDelegate;
+
 public class KNN920CodecTests extends KNNCodecTestCase {
 
     public void testMultiFieldsKnnIndex() throws Exception {
-        testMultiFieldsKnnIndex(new KNN920Codec());
+        testMultiFieldsKnnIndex(KNN920Codec.builder().delegate(createKNN92DefaultDelegate()).build());
     }
 
     public void testBuildFromModelTemplate() throws InterruptedException, ExecutionException, IOException {
-        testBuildFromModelTemplate(new KNN920Codec());
+        testBuildFromModelTemplate((KNN920Codec.builder().delegate(createKNN92DefaultDelegate()).build()));
+    }
+
+    public void testKnnVectorIndex() throws Exception {
+        MapperService mapperService = mock(MapperService.class);
+        final KNN920Codec actualCodec = KNN920Codec.builder()
+            .delegate(createKNN92DefaultDelegate())
+            .mapperService(Optional.of(mapperService))
+            .build();
+        final KNN920Codec codec = KNN920Codec.builder()
+            .delegate(createKNN92DefaultDelegate())
+            .mapperService(Optional.of(mapperService))
+            .build();
+        setUpMockClusterService();
+        Directory dir = newFSDirectory(createTempDir());
+        IndexWriterConfig iwc = newIndexWriterConfig();
+        iwc.setMergeScheduler(new SerialMergeScheduler());
+        iwc.setCodec(codec);
+
+        /**
+         * Add doc with field "test_vector"
+         */
+        FieldType luceneFieldType = new FieldType(KNNVectorFieldMapper.Defaults.FIELD_TYPE);
+        luceneFieldType.putAttribute(KNNConstants.KNN_METHOD, KNNConstants.METHOD_HNSW);
+        luceneFieldType.putAttribute(KNNConstants.KNN_ENGINE, KNNEngine.LUCENE.getName());
+        luceneFieldType.putAttribute(KNNConstants.SPACE_TYPE, SpaceType.L2.getValue());
+        luceneFieldType.putAttribute(KNNConstants.HNSW_ALGO_M, "32");
```

Comment:
Are we able to test that these get set correctly?

Reply:
Good catch, it wasn't actually. Changed test so it uses correct fieldtype